backends/ref/ceed-ref.h

```diff
@@ -30,8 +30,7 @@ typedef struct {
   // Orientation, if it exists, is true when the face must be flipped (multiplies by -1.).
   const bool *orient;
   bool       *orient_allocated;
-  int (*Apply)(CeedElemRestriction, const CeedInt, const CeedInt, const CeedInt, CeedInt, CeedInt, CeedTransposeMode, CeedVector, CeedVector,
-               CeedRequest *);
+  int (*Apply)(CeedElemRestriction, CeedInt, CeedInt, CeedInt, CeedInt, CeedInt, CeedTransposeMode, CeedVector, CeedVector, CeedRequest *);
```

Comment:
I must be missing something here so I am looking for some help, but why does this work when the definition for `CeedElemRestrictionApply_Ref_Core` is still using the previous signature with `const CeedInt`s in `backends/ref/ceed-ref-restriction.c`?

Reply:
@jeremylt can probably explain it better than I can, but I believe the `const` in the function definition only applies to within the function context itself. Since the `CeedInt` parameters are pass-by-value, having a `const` on their signature is meaningless; the value is copied and independent of the variable passed to the function when called. This [SO](https://softwareengineering.stackexchange.com/a/204720/373632) answer covers it fairly well I think.

Reply:
This makes sense, `CeedInt` -> `const CeedInt` is a valid compatible conversion on call to the function. Thanks!